src/test/hosting/api.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { expect } from "chai";
+import * as nock from "nock";
+import * as sinon from "sinon";
+import * as helpers from "../helpers";
+import * as api from "../../api";
+import * as hostingApi from "../../hosting/api";
+
+const TEST_CHANNELS_RESPONSE = {
+  channels: [
+    // domain exists in TEST_GET_DOMAINS_RESPONSE
+    { url: "https://my-site--ch1-4iyrl1uo.web.app" },
+    // domain does not exist in TEST_GET_DOMAINS_RESPONSE
+    { url: "https://my-site--ch2-ygd8582v.web.app" },
+  ],
+};
+const TEST_GET_DOMAINS_RESPONSE = {
+  authorizedDomains: [
+    "my-site.firebaseapp.com",
+    "localhost",
+    "randomurl.com",
+    "my-site--ch1-4iyrl1uo.web.app",
+    // domain that should be removed
+    "my-site--expiredchannel-difhyc76.web.app",
+  ],
+};
+
+const EXPECTED_DOMAINS_RESPONSE = [
+  "my-site.firebaseapp.com",
+  "localhost",
+  "randomurl.com",
+  "my-site--ch1-4iyrl1uo.web.app",
+];
+const PROJECT_ID = "test-project";
+const SITE = "my-site";
+
+describe("hosting", () => {
+  beforeEach(() => {
+    helpers.mockAuth(sinon);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe("getApprovedDomains", () => {
+    afterEach(() => {
+      nock.cleanAll();
+    });
+
+    it("it should return the list of expetecd auth domains after syncing", async () => {
+      // mock listChannels response
+      nock(api.hostingApiOrigin)
+        .get(`/v1beta1/projects/${PROJECT_ID}/sites/${SITE}/channels`)
+        .query((queryParams: any) => {
```

Comment:
I don't recall why you need this `.query` - does `nock` filter on query parameters (and this wouldn't work without specifying?)?

Reply:
yes because the for the mock to work it need to match the real query 100%